### PR TITLE
[US013,US014] Dying and respawn

### PR DESCRIPTION
**Pull request by krzysiekprzekwas**

### CODE REVIEW

File: GameServer/Game/SpawnService.cs [hunk 1]
```diff
@@ -20,6 +20,10 @@ public static void SpawnPlayer(Player player)
 
             var weaponEnum = WeaponService.GetDefaultWeaponEnum();
             player.PlayerWeapon = new PlayerWeapon(weaponEnum);
+            
+            player.IsAlive = true;
```
Comment (tomejzen) [inline GameServer/Game/SpawnService.cs]: Where is health property changed to maximum?

File: GameServer/wwwroot/index.html [hunk 2]
```diff
@@ -10,15 +10,24 @@
...
-    </span>
+    <span id="healthLabel" class="measurementLabel no-select"></span>
+<span id="fullscreenToggle" class="no-select">
+    <img src="#" alt="" />
+</span>
+
+<div id="killScreen" class="hidden">
+    <div class="center">
+        <p class="killText">You  died!</p>
+        <p>Yoi'll be revived in </p>
```
Comment (tomejzen) [inline GameServer/wwwroot/index.html]: Yoi'x

File: GameServer/wwwroot/style/main.css [hunk 3]
```diff
@@ -62,3 +62,34 @@ body {
 #healthLabel {
     top: calc(8px + 2rem);
 }
+
+.overlay {
```
Comment (tomejzen) [inline GameServer/wwwroot/style/main.css]: Maybe we can add here:
user-select: none

Maybe in other classes too?

File: GameServer/Hubs/GameHub.cs [hunk 4]
```diff
@@ -63,12 +63,12 @@ public void ClientStateUpdate(string clientState)
 
             var player = _gameEngine.GameState.Players.Find(x => x.ConnectionId == Context.ConnectionId);
 
-            if (player==null)
+            // Ignore update from dead
+            if (!player.IsAlive)
```
Comment (tomejzen) [inline GameServer/Hubs/GameHub.cs]: Don't we still need player == null?

File: GameServer/Game/SpawnService.cs (hunk 1 above)
Comment (krzysiekprzekwas) [reply GameServer/Game/SpawnService.cs]: In GameEngine. Spawn service had no Config injection. I can Change it if U want

File: GameServer/wwwroot/index.html (hunk 2 above)
Comment (krzysiekprzekwas) [reply GameServer/wwwroot/index.html]: Daa

File: GameServer/Hubs/GameHub.cs (hunk 4 above)
Comment (krzysiekprzekwas) [reply GameServer/Hubs/GameHub.cs]: should be, will adress this

File: GameServer/Game/GameEngine.cs [hunk 5]
```diff
@@ -100,6 +112,16 @@ private void ApplyDamage()
...
                 {
                     colidedPlayer.Health -= _gameState.Players.First(x => x.Id == bullet.PlayerId).PlayerWeapon.GetWeapon().BulletDamage;
+
+                    if (colidedPlayer.Health <= 0)
+                    {
+                        colidedPlayer.IsAlive = false;
+                        colidedPlayer.MilisecondsToResurect = _config.MilisecondsToResurect;
+
+                        var killer = _gameState.Players.First(x => x.Id == bullet.PlayerId);
+                        _hubContext.Clients.All.SendAsync("playerKilled",new object[]{ killer.Name, colidedPlayer.Name });
```
[the remark this answers is not in the document]
Comment (krzysiekprzekwas) [reply GameServer/Game/GameEngine.cs]: NO WE DO NOT!

**Pull request merged**